controllers/componentdetectionquery_controller.go

```diff
@@ -247,7 +246,7 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 		}
 
 		for context, link := range dockerfileContextMap {
-			updatedLink, err := devfile.UpdateGitLink(source.URL, source.Revision, link)
+			updatedLink, err := devfile.UpdateGitLink(source.URL, source.Revision, path.Join(context, link))
```

Comment:
Do we need to add the context to the path?  For multicomponents, it looks like it's already done in the search function https://github.com/redhat-appstudio/application-service/blob/main/pkg/devfile/detect.go#L108

Reply:
So the reason I added that line is that believe it's two separate contexts. The context here is the GitSource context that we want to use as the base folder for scanning for devfiles (defaults to `./` but can be anything). The context in your link is the local path under the scanned folder that contains a devfile (e.g. `component-a`)

So pretend we have a Git repo like so:
```
dev/
   component-a/devfile.yaml
   component-b/
   component-c/
somefolder/
anotherfolder/
```

So you would run the CDQ with `GitSource.context = "dev"`, and you'd want the returned devfile URLs to be https://raw.githubusercontent.com/myorg/myrepo/main/dev/component-a/devfile.yaml

Reply:
Unresolving this as i had a question, what happens when detect.go returns a absolute path rather than something relative

Reply:
Updated to handle absolute paths (i.e. http) properly